### PR TITLE
refactor: separate npm extract functions

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Refactors npm extract functions

## Context

Preparation for npm manager separation + bun

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/npm/extract/__snapshots__/index.spec.ts.snap [hunk 1]
```diff
@@ -131,7 +131,6 @@ exports[`modules/manager/npm/extract/index .extractPackageFile() extracts engine
     "npm": "^8.0.0",
     "pnpm": "^1.2.0",
     "vscode": ">=1.49.3",
-    "yarn": "disabled",
```
Comment (rarkins) [inline lib/modules/manager/npm/extract/__snapshots__/index.spec.ts.snap:134]: The automatic suppression of this is intentional

**Comment by rarkins** [comment]
I'm using this branch to implement bun support off, and may still have some changes before then. This approach will be helpful for the later npm manager separation

**Comment by HonkingGoose** [comment]
> I'm using this branch to implement bun support off, and may still have some changes before then. This approach will be helpful for the later npm manager separation

See:

- #5607

**Review by viceice — DISMISSED** [review]

File: lib/modules/manager/npm/extract/index.ts [hunk 2]
```diff
@@ -501,16 +196,31 @@ export async function extractPackageFile(
     }
   }
 
+  const extractedConstraints = getExtractedConstraints(res.deps);
+
+  if (yarnConfig) {
+    for (const dep of res.deps) {
+      if (dep.depName) {
```
[the remark this answers is not in the document]
Comment (rarkins) [reply lib/modules/manager/npm/extract/index.ts:203]: Yes, I think so

**Review by viceice — APPROVED** [review]

**Pull request merged**